### PR TITLE
Replace FindcuFile with upstream FindCUDAToolkit support

### PR DESCRIPTION
**Pull request by KyleFromNVIDIA**
## Description
CMake's `FindCUDAToolkit` has supported cuFile since 3.25. Use this support and remove the custom `FindcuFile` module.

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

**Review by bdice — APPROVED** [review]

File: cpp/CMakeLists.txt [hunk 1]
```diff
@@ -902,6 +897,11 @@ target_compile_definitions(cudf PUBLIC "SPDLOG_ACTIVE_LEVEL=SPDLOG_LEVEL_${LIBCU
 # Enable remote IO through KvikIO
 target_compile_definitions(cudf PRIVATE $<$<BOOL:${CUDF_KVIKIO_REMOTE_IO}>:CUDF_KVIKIO_REMOTE_IO>)
 
+# Enable cuFile support
+if(TARGET CUDA::cuFile)
+  target_compile_definitions(cudf PRIVATE CUFILE_FOUND)
```
Comment (bdice) [inline cpp/CMakeLists.txt:902]: Could we rename this to `CUDF_CUFILE_FOUND`? We have the name `KVIKIO_CUFILE_FOUND` in kvikio.

Comment (KyleFromNVIDIA) [reply cpp/CMakeLists.txt:902]: I've renamed it locally, will push once CI passes.

**Review by bdice — APPROVED** [review]
Approving C++ changes now.

**Review by karthikeyann — APPROVED** [review]

**Review by PointKernel — APPROVED** [review]

File: cpp/src/io/utilities/file_io_utilities.hpp [hunk 2]
```diff
@@ -97,7 +97,7 @@ class cufile_output {
   virtual std::future<void> write_async(void const* data, size_t offset, size_t size) = 0;
 };
 
-#ifdef CUFILE_FOUND
+#ifdef CUDF_CUFILE_FOUND
```
Comment (vuule) [inline cpp/src/io/utilities/file_io_utilities.hpp:100]: can't wait to remove these as well :)

**Review by mhaseeb123 — APPROVED** [review]

**Pull request merged**